### PR TITLE
circular buffer test case fix

### PR DESCRIPTION
**Pull request by lbblack**
In the util method, read_values_from_buffer, read value is set to zero and fails the assertion tests.

### CODE REVIEW

File: exercises/circular-buffer/test/test_circular_buffer.c [hunk 1]
```diff
@@ -36,7 +36,7 @@ static void read_values_from_buffer(size_t length, buffer_value_t values[],
                                     circular_buffer_t * buffer)
 {
    for (size_t i = 0; i < length; ++i) {
-      buffer_value_t read_value = 0;
+      buffer_value_t read_value = values[i];
```
Comment (ryanplusplus) [inline exercises/circular-buffer/test/test_circular_buffer.c:39]: Since `read_value` should be assigned during the `read` all on the next line, why is this change necessary? I'm having a hard time understanding why we need an initial value at all and why `value[i]` is better than any other value.

Comment (wolf99) [reply exercises/circular-buffer/test/test_circular_buffer.c:39]: Agree with @ryanplusplus, this change would cause even an empty `read()` function to pass the tests. I'm not sure what is trying to be achieved by this change.

**Comment by patricksjackson** [comment]
Closing, since this looks to just be a misunderstanding.
